packages/code-du-travail-frontend/src/outils/IndemniteLicenciement/ccn/1486/steps/AncienneteETAMIC.js

```diff
@@ -0,0 +1,200 @@
+import React from "react";
+import styled from "styled-components";
+import { Field } from "react-final-form";
+import { FieldArray } from "react-final-form-arrays";
+import { OnChange } from "react-final-form-listeners";
+import { Toast, Button, Table, theme, Wrapper } from "@cdt/ui";
+
+import { BlockError, Input, SectionTitle } from "../../../stepStyles";
+import { YesNoQuestion } from "../../../components/YesNoQuestion";
+import { isNumber, isRequiredNumber } from "../../../validators";
+
+export const CONTRATS_KEY = "brancheContrats";
+export const HAS_CONTRATS_KEY = "hasBrancheContrat";
+
+export const AncienneteETAMIC = ({ form }) => {
+  const data = form.getState().values;
+  const hasContrats = data[HAS_CONTRATS_KEY];
+
+  return (
+    <>
+      <SectionTitle>Contrats antérieurs</SectionTitle>
+      <Wrapper variant="dark">
+        <p>
+          Si vous avez été auparavant employé par la même société et que la
+          résiliation du/des contrat(s) n’est pas imputable à l’une des raisons
+          suivantes:
+        </p>
+        <ul>
+          <li>démission</li>
+          <li>licenciement pour faute grave</li>
+          <li>interruption pour incapacité de travail inférieure à 6 mois</li>
+          <li>détachement auprès d’une filiale</li>
+        </ul>
+        <p>
+          alors{" "}
+          <strong>
+            la durée de ce(s) contrat(s) s’ajoute à votre ancienneté
+          </strong>
+          .
+        </p>
+      </Wrapper>
+      <YesNoQuestion
+        name={HAS_CONTRATS_KEY}
+        label="Avez-vous un/des contrat(s) antérieur(s) à déclarer ?"
+        onChange={HAS_CONTRATS_KEY => {
+          HAS_CONTRATS_KEY
+            ? form.change(CONTRATS_KEY, [{}])
+            : form.change(CONTRATS_KEY, []);
+        }}
+      />
+      {hasContrats && (
```

Comment:
You can use a field here to conditionnaly get a form value use ask for
```jsx
<Field name={HAS_CONTRATS_KEY}>{({input}) => {input.value === true && (<Jsx/>)}</Field>
```

Reply:
that way you don't need a use the form props

Reply:
I find it very obscure to use a Field in order to have a conditionnal rendering

Reply:
it's used in many place here and there (it's use for error handling, for conditional viewing)

Reply:
I'm not sure that counterweight the amount of additional boilerplate this adds. It's really confusing to me.

Reply:
because we could refactor it later to have a <dedicated> component to do that

Reply:
please...  also it don't enforce to passes form props

Reply:
I'm not sure I understand the logic behind that ? You mean you need to add the Field component because later you might need to wrap it in another component ?? Do you have an example I might grasp better ?

Reply:
My point was mainly about having uniform practice and way of doing stuff. I don't want to discuss that point more than it needs, I'll resolve the conversation. You can leave the piece of code as it is.